server/auth/github/client_test.go

```diff
@@ -1,3 +1,8 @@
+// Disabled based on https://github.com/remind101/empire/commit/775178497a2ce27a5798a54937dea256c6253dc9
+// "it seems like the mock library cannot detect our functions for
+// any go version over 1.7.6 -- this comment will disable these tests."
+// +build !go1.10
```

Comment:
Context for why these tests are disabled:

As best I can tell, later versions of golang introduced a new attribute of the `http.Request` object which is a `GetBody` function. This made it so you can't compare the equality of two `http.Request`s, because in go you can't compare functions.

The test failures manifest as "unexpected method call" errors:
```
mock: Unexpected Method Call
-----------------------------

Do(*http.Request)
		0: &{POST https://api.github.com/authorizations HTTP/1.1 1 1 map[Accept:[application/vnd.github.v3+json] Authorization:[Basic dXNlcm5hbWU6cGFzc3dvcmQ=]] {{"scopes":["scope"],"client_id":"client_id","client_secret":"client_secret"}
} 0x14b10c0 77 [] false api.github.com map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000124008}

The closest call I have is:

Do(*http.Request)
		0: &{POST https://api.github.com/authorizations HTTP/1.1 1 1 map[Accept:[application/vnd.github.v3+json] Authorization:[Basic dXNlcm5hbWU6cGFzc3dvcmQ=]] {{"scopes":["scope"],"client_id":"client_id","client_secret":"client_secret"}
} 0x14b10c0 77 [] false api.github.com map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000124008}
```
These happen on every single test in this file that asserts that the client's `Do` method has been called with a particular `http.Request`, which is all or most of the tests.

I started looking into how to mock objects in go and/or how to assert things about an object a mock is called with, but didn't get far yet. Very interested in any leads anyone has so we can get these tests running!

Reply:
I don't have many leads myself unfortunately. My naive guess is that we may have to explicitly assert the inputs in a mock function, but that's just a guess.